src/main/java/com/openlattice/chronicle/services/edm/EdmCacheService.java

```diff
@@ -0,0 +1,242 @@
+        appNameIdMap.forEach( ( appComponent, appId ) -> {
+            List<UserAppConfig> configs = appApi.getAvailableAppConfigs( appId );
+
+            if ( configs.isEmpty() )
+                return;
+
+            // get EntityTypeCollection associated with app
+            EntitySetCollection entitySetCollection = collectionsApi
+                    .getEntitySetCollection( configs.get( 0 ).getEntitySetCollectionId() );
+            EntityTypeCollection entityTypeCollection = collectionsApi
+                    .getEntityTypeCollection( entitySetCollection.getEntityTypeCollectionId() );
+
+            // create mapping from templateTypeName -> templateTypeId
+            Map<String, UUID> templateTypeNameIdMap = entityTypeCollection
+                    .getTemplate()
+                    .stream()
+                    .collect( Collectors.toMap( CollectionTemplateType::getName, AbstractSecurableObject::getId ) );
+
+            // for each config map orgId -> templateTypeName -> ESID
+            Map<UUID, Map<CollectionTemplateTypeName, UUID>> orgEntitySetMap = new HashMap<>();
+
+            configs.forEach( userAppConfig -> {
+                Map<UUID, UUID> templateTypeIdESIDMap = collectionsApi
+                        .getEntitySetCollection( userAppConfig.getEntitySetCollectionId() ).getTemplate();
+
+                // iterate over templateTypeName enums and create mapping templateTypeName -> entitySetId
+                Map<CollectionTemplateTypeName, UUID> templateTypeNameESIDMap = Maps
+                        .newHashMapWithExpectedSize( templateTypeIdESIDMap.size() );
+
+                for ( CollectionTemplateTypeName templateTypeName : CollectionTemplateTypeName.values() ) {
+                    UUID templateTypeId = templateTypeNameIdMap.get( templateTypeName.toString() );
+
+                    if ( templateTypeId != null ) {
+                        templateTypeNameESIDMap.put( templateTypeName, templateTypeIdESIDMap.get( templateTypeId ) );
+                    }
+                }
+                orgEntitySetMap.put( userAppConfig.getOrganizationId(), templateTypeNameESIDMap );
+            } );
+
+            entitySets.put( appComponent, orgEntitySetMap );
+        } );
+
+        entitySetIdsByOrgId.putAll( entitySets );
+    }
+
+    @Override
+    public UUID getPropertyTypeId( FullQualifiedName fqn ) {
+        return this.propertyTypeIdsByFQN.get( fqn );
+    }
+
+    @Override
+    public PropertyType getPropertyType( UUID propertyTypeId ) {
+        return propertyTypesById.get( propertyTypeId );
+    }
+
+    @Override
+    public PropertyType getPropertyType( FullQualifiedName fqn ) {
+        return getPropertyType( getPropertyTypeId( fqn ) );
+    }
+
+    @Override
+    public Map<String, UUID> getPropertyTypeIds( Set<String> propertyTypeFqns ) {
+        EdmApi edmApi;
+        try {
+            ApiClient apiClient = apiCacheManager.prodApiClientCache.get( ApiClient.class );
```

Comment:
Correct. However, I still think we should not the endpoint's return type and input because of https://github.com/openlattice/chronicle-server/pull/143#discussion_r518325261

Though I suppose we could retain the current endpoint as legacy and create a new one which is more aptly typed. Thoughts?